### PR TITLE
Prefinal PR

### PR DESCRIPTION
**Pull request by raysandeep**
- [x] Record Audio
- [x] Upload to S3
- [ ] Convert Ogg to Mp3 
- [x] Lambda Trigger to convert to Text
- [ ] Text Summarization

### CODE REVIEW

File: record_audio.go [hunk 1]
```diff
@@ -1,61 +1,65 @@
...
 	"github.com/pion/webrtc/v3/pkg/media/oggwriter"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
 )
 
-// Variables used for command line parameters
+var ctx = context.Background()
```
Comment (L04DB4L4NC3R) [inline record_audio.go]: You can use a single var block for global definitions

File: record_audio.go [hunk 2]
```diff
@@ -1,61 +1,65 @@
...
+	LoadEnv()
+	Token := GetEnvWithKey("Token")
+	var err error
+	dgVoice, err = discordgo.New("Bot " + Token)
+	if err != nil {
+		log.Println("error creating Discord session,", err)
+		return
 	}
-	return string(b)
+	defer dgVoice.Close()
```
Comment (L04DB4L4NC3R) [inline record_audio.go:37]: Why is the close() coming before the open?

File: record_audio.go [hunk 3]
```diff
@@ -224,3 +171,58 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
 	}
 
 }
+
+
+
+
+//AWS - S3
+
+func ConnectAws() *session.Session {
```
Comment (L04DB4L4NC3R) [inline record_audio.go]: Have these function definitions in a different file. All your code is in 1 file mainly

File: record_audio.go [hunk 4]
```diff
@@ -224,3 +171,58 @@ func messageCreate(s *discordgo.Session, m *discordgo.MessageCreate) {
...
+			),
+		})
+	if err != nil {
+		log.Println("Error :",err)
+	}
+	return sess
+}
+
+
+func AddFileToS3(s *session.Session, fileDir string) error {
```
Comment (L04DB4L4NC3R) [inline record_audio.go]: Everythibng S3 related in separate file

File: record_audio.go [hunk 5]
```diff
@@ -1,61 +1,65 @@
...
 	"github.com/aws/aws-sdk-go/aws/session"
-	"github.com/aws/aws-sdk-go/service/dynamodb"
-	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
 	"github.com/bwmarrin/discordgo"
 	"github.com/joho/godotenv"
 	"github.com/pion/rtp"
 	"github.com/pion/webrtc/v3/pkg/media"
 	"github.com/pion/webrtc/v3/pkg/media/oggwriter"
+	"log"
```
Comment (L04DB4L4NC3R) [inline record_audio.go:11]: Use Logrus

**Review by L04DB4L4NC3R — CHANGES_REQUESTED** [review]

File: record_audio.go (hunk 2 above)
Comment (raysandeep) [reply record_audio.go:37]: Defer anyways close voice connection when we are killing the bot na.

File: record_audio.go (hunk 3 above)
Comment (raysandeep) [reply record_audio.go]: Will do these changes.

**Review by raysandeep — COMMENTED** [review]
Made all changes.

**Review by L04DB4L4NC3R — APPROVED** [review]

**Pull request merged**